### PR TITLE
feat: add RegexNodeType

### PR DESCRIPTION
**Pull request by Le-Pepp**
Gives a list of groups rather than dynamically outputting each group.

### CODE REVIEW

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RegexNodeType.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    @Override
+    public Map<String, CompletableFuture<DataBox<?>>> compute(Map<String, DataBox<?>> inputs, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {        
+        var regex = Pattern.compile(DataBox.get(inputs, "regex", DataType.STRING).orElseThrow());        
+        var word = DataBox.get(inputs, "word", DataType.STRING).orElseThrow();
+        var matcher = regex.matcher(word);
+
+        if (!matcher.find()) return Map.of(
+            "isMatch", CompletableFuture.completedFuture(DataType.BOOLEAN.create(false)),
+            "match", CompletableFuture.completedFuture(DataType.STRING.create(null)),
+            "groups", CompletableFuture.completedFuture(DataType.STRING.listOf().create(null))
```
Comment (williambl) [inline basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RegexNodeType.java:65]: We never return nulls from nodes. If there's a possibility that something could not have an output, use an optional type.

**Review by williambl — CHANGES_REQUESTED** [review]
Nice docs.

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RegexNodeType.java (hunk 1 above)
Comment (Le-Pepp) [reply basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/RegexNodeType.java:65]: https://user-images.githubusercontent.com/61510644/226700080-30931186-9331-4049-9aef-43323570bfb6.mp4

**Pull request merged**